src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Implementation/Policy/NewAzurePolicyAssignment.cs

```diff
@@ -112,11 +144,73 @@ private JToken GetResource()
                 {
                     DisplayName = this.DisplayName ?? null,
                     PolicyDefinitionId = this.PolicyDefinition.Properties["policyDefinitionId"].Value.ToString(),
-                    Scope = this.Scope
+                    Scope = this.Scope,
+                    Parameters = this.GetParameters()
                 }
             };
 
             return policyassignmentObject.ToJToken();
         }
+
+        object IDynamicParameters.GetDynamicParameters()
+        {
+            if (this.PolicyDefinition != null)
+            {
+                var properties = this.PolicyDefinition.Properties["Properties"];
+                if (properties != null)
+                {
+                    var parameters = (properties.Value as PSObject).Properties["parameters"];
+                    if (parameters != null)
+                    {
+                        foreach (var param in (parameters.Value as PSObject).Properties)
+                        {
+                            var type = (param.Value as PSObject).Properties["type"];
+                            var typeString = type != null ? type.Value.ToString() : string.Empty;
+                            var dp = new RuntimeDefinedParameter
+                            {
+                                Name = param.Name,
+                                ParameterType = typeString.Equals("array", StringComparison.OrdinalIgnoreCase) ? typeof(object[]) : typeof(string)
+                            };
+                            dp.Attributes.Add(new ParameterAttribute
+                            {
+                                Mandatory = false,
+                                ValueFromPipelineByPropertyName = true,
```

Comment:
@cormacpayne Similarly, the user can put dynamic parameters in an object (say $param) and say: $param | New-AzureRmPolicyAssignment